src/components/Tag/Tag.vue

```diff
@@ -0,0 +1,100 @@
+<script lang="ts" setup>
+import { ref, defineProps, defineEmits, useSlots } from 'vue';
+import { TagProps } from './types';
+
+const props = withDefaults(defineProps<TagProps>(), {
+  size: 'm',
+  style: 'outlined',
+  disabled: false,
+  clickable: false,
+  removable: false,
+});
+
+const slots = useSlots();
+let containerClasses = '';
+const hasContent = slots.default;
+const hasSymbol = props.icon || props.color;
+const hasAction = props.removable || props.expandable;
+
+if (props.size === 's') {
+  containerClasses += '  text-xs h-5 min-w-[20px] py-0.5 rounded ';
+
+  // Variant: SYMBOL-TEXT
+  if (hasSymbol && hasContent && !hasAction) {
+    containerClasses += 'pr-1 ';
+  }
+  // Variant: TEXT-ACTION
+  else if (!hasSymbol && hasContent && hasAction) {
+    containerClasses += 'pl-1 ';
+  }
+
+  // Variant: TEXT
+  if (hasContent) {
+    containerClasses += 'px-1 ';
+  }
+} else if (props.size === 'm') {
+  containerClasses += ' text-sm h-6 min-w-[24px] leading-4 p-1 rounded-md ';
+
+  // Variant: SYMBOL-TEXT
+  if (hasSymbol && hasContent && !hasAction) {
+    containerClasses += 'pr-2 ';
+  }
+  // Variant: TEXT-ACTION
+  else if (!hasSymbol && hasContent && hasAction) {
+    containerClasses += 'pl-2 ';
+  }
+  // Variant: TEXT
+  else if (!hasSymbol && hasContent && !hasAction) {
+    containerClasses += 'px-2 ';
+  }
+}
+const allStyleClasses = {
+  ghost: 'outline-transparent',
+  outlined: 'outline-slate-500',
+};
+const emits = defineEmits(['action', 'click']);
+const disabledClasses = props.disabled ? 'opacity-50' : '';
+const disabledActionClasses = props.disabled ? 'cursor-default' : 'hover:bg-slate-400';
+const clickableClasses = props.clickable ? 'cursor-pointer hover:bg-slate-300 outline-slate-300' : '';
```

Comment:
If I have set `disabled` and `clickable` both to true, the UI will show `cursor-pointer` right?